src/main/minio.js

```diff
@@ -1405,35 +1407,47 @@ export class Client {
   //   * `stat.etag` _string_: etag of the object
   //   * `stat.metaData` _string_: MetaData of the object
   //   * `stat.lastModified` _Date_: modified time stamp
-  statObject(bucketName, objectName, cb) {
+  statObject(bucketName, objectName, statOpts={}, cb) {
```

Comment:
@ebozduman i have used GO sdk's [StatObject](https://docs.min.io/docs/golang-client-api-reference.html#StatObject) as a reference.
  the `statOpts` object is used here because it provides similar way of passing parameters  to API like other SDKs. also. since the last argument needs to be a callback , adding backward compatibility can be difficult as we need to keep adding if/else to shift the arguments. as we enhance the apis to support more S3 like features if any.

 the AWS signature had just 2 arguments.
`headObject(params = {}, callback)`

so i took this approach.

Any thoughts?

Reply:
Then it is good.
If the minio-go implementation is this way, then it means there are some other possible parameters we may support in the future.